common/src/python/preprocess/preprocessor.py

```diff
@@ -996,6 +887,118 @@ def is_existing_visit(self, *, input_record: Dict[str, Any]) -> bool:
+                return True
+
+        # no clinical visits found; fail preprocessing check
+        self.__error_handler.write_module_error(
+            pp_context=pp_context, error_code=SysErrorCodes.CLINICAL_FORM_REQUIRED
+        )
+
+        return False
+
+    def _check_np_mlst_restrictions(self, pp_context: PreprocessingContext) -> bool:
+        """Check NP/MLST restrictions; compares NP form against most recent
+        MLST form.
+
+        Args:
+            pp_context: preprocessing context
+        Returns:
+            True if the preprocessing checks pass, false otherwise
+        """
+        if self.__module != DefaultValues.NP_MODULE:
+            raise PreprocessingException(
+                "Cannot evaluate NP/MLST preprocessing "
+                + f"checks for non-NP module: {self.__module}"
+            )
+
+        assert pp_context.subject_lbl, "pp_context.subject_lbl required"
+
+        # get most recent MLST form
+        all_mlst_forms = self.__forms_store.query_form_data(
+            subject_lbl=pp_context.subject_lbl,
+            module=DefaultValues.MLST_MODULE,
+            legacy=False,
+            search_col=FieldNames.DATE_COLUMN,
+            find_all=True,
+            extra_columns=["deathyr", "deathmo", "deathdy", "autopsy"],
+        )
+
+        # if no MLST forms, passes by default
+        if not all_mlst_forms:
+            return True
+
+        mlst_form = all_mlst_forms[0]
+        input_record = pp_context.input_record
+
+        # preprocess-026: death dates in MLST/NP must match
+        mlst_year = mlst_form.get(f"{MetadataKeys.FORM_METADATA_PATH}.deathyr")
+        np_year = input_record.get("npdodyr")
+
+        mlst_dod = build_date(
+            year=mlst_year,
+            month=mlst_form.get(f"{MetadataKeys.FORM_METADATA_PATH}.deathmo"),
+            day=mlst_form.get(f"{MetadataKeys.FORM_METADATA_PATH}.deathdy"),
+        )
+        np_dod = build_date(
+            year=np_year,
+            month=input_record.get("npdodmo"),
+            day=input_record.get("npdoddy"),
+        )
+
+        result_dod = True
+        # MLST may have not had month/day defined (unknown 99s)
+        # if so, compare year directly
+        # otherwise check both exist and are equal to each other
+        if not mlst_dod and mlst_year is not None:
+            result_dod = mlst_year == np_year
+        else:
+            result_dod = (mlst_dod is not None and np_dod is not None) and (
+                mlst_dod == np_dod
+            )
+
+        if not result_dod:
+            self.__error_handler.write_date_error(
+                pp_context=pp_context,
+                error_code=SysErrorCodes.DEATH_DATE_MISMATCH,
+                date_field="npdodyr",
+            )
+
+        # preprocess-027: autopsy must be 1 in MLST form for NP to be accepted
+        autopsy = mlst_form.get(f"{MetadataKeys.FORM_METADATA_PATH}.autopsy", "")
```

Comment:
Let's move this check before DOD check. I asked RT to switch this to error 026 and DOD check to 027, and add "deceased" == 1 check for this error code.